care/facility/tests/test_ambulance_api.py

```diff
@@ -0,0 +1,298 @@
+from typing import Any
+
+from rest_framework import status
+
+from care.facility.models import Ambulance, AmbulanceDriver, District
+from care.utils.tests.test_base import TestBase
+from config.tests.helper import mock_equal
+
+
+class TestAmbulance(TestBase):
+    """Test ambulance APIs"""
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Runs once per class
+            - Initialize the attributes useful for class methods
+        """
+        super(TestAmbulance, cls).setUpClass()
+
+    def get_base_url(self):
+        return "/api/v1/ambulance"
+
+    def get_detail_representation(self, ambulance: Any):
+
+        if isinstance(ambulance, dict):
+            this_ambulance = ambulance.copy()
+            this_ambulance.pop("drivers", [])
+            this_ambulance_data = this_ambulance.copy()
+            this_ambulance_data["primary_district"] = self.get_district_object(this_ambulance_data["primary_district"])
+            this_ambulance_data["secondary_district"] = self.get_district_object(
+                this_ambulance_data["secondary_district"]
+            )
+            this_ambulance_data["third_district"] = self.get_district_object(this_ambulance_data["third_district"])
+            ambulance = Ambulance(**this_ambulance_data)
+
+        return {
+            "id": mock_equal,
+            "drivers": mock_equal,
+            "primary_district_object": self.format_district(ambulance.primary_district),
+            "secondary_district_object": self.format_district(ambulance.secondary_district),
+            "third_district_object": self.format_district(ambulance.third_district),
+            "created_date": mock_equal,
+            "modified_date": mock_equal,
+            "deleted": ambulance.deleted,
+            "vehicle_number": ambulance.vehicle_number,
+            "owner_name": ambulance.owner_name,
+            "owner_phone_number": ambulance.owner_phone_number,
+            "owner_is_smart_phone": ambulance.owner_is_smart_phone,
+            "has_oxygen": ambulance.has_oxygen,
+            "has_ventilator": ambulance.has_ventilator,
+            "has_suction_machine": ambulance.has_suction_machine,
+            "has_defibrillator": ambulance.has_defibrillator,
+            "insurance_valid_till_year": ambulance.insurance_valid_till_year,
+            "ambulance_type": ambulance.ambulance_type,
+            "price_per_km": ambulance.price_per_km,
+            "has_free_service": ambulance.has_free_service,
+            "primary_district": ambulance.primary_district.id,
+            "secondary_district": ambulance.secondary_district.id,
+            "third_district": ambulance.third_district.id,
+        }
+
+    def get_district_object(self, dist_obj_id):
+        return District.objects.get(id=dist_obj_id)
+
+    def format_district(self, district):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5398997)